Keess v0.1.0

mvleandro opened the pull request:
### PR Description
This PR creates a new chart for the new secrets and configmaps synchronizer named keess.

### Checklist

 - [X] Have you reviewed and updated the chart default values if necessary?
 - [X] Have you reviewed and updated the chart documentation if necessary?
 - [X] Does your branch follow the naming convention of `{chartNameWithDashes}-v{versionString}-{optionalPatchVersion}`?
 - [X] Have you bumped the version in the chart's `Chart.yaml`?

### Tagged Releases
Please remember to make a tagged release after merging your PR that:

 - Has a tag name that matches your PR branch name (see above)
 - Has a description that summarizes the changes made

This makes it possible to use previous versions of the charts maintained here as new releases are published. Please see the release history of this repository for examples.

mvleandro committed:
Creating a helm chart to deploy keess.

mvleandro committed:
Adding missing command-line arguments.

mvleandro committed:
Changing secret variable name.

mvleandro committed:
Changing variable name on secret template.

mvleandro committed:
Removing conditional volume mount.

mvleandro committed:
Fixing volumes identation on deployment.

mvleandro committed:
Fixing volumes identation on deployment.

mvleandro committed:
Fixing typo on NOTES.txt.

mvleandro committed:
Adding namespace template.

iblackman commented on charts/keess/templates/secret.yaml:
Do you think you will use this?

mvleandro replied:
> Do you think you will use this?

Done!

mvleandro replied:
> Do you think you will use this?

Yes!

iblackman commented on charts/keess/templates/deployment.yaml:
```diff
@@ -0,0 +1,77 @@
+          - name: config
+            mountPath: /root/.kube
+            readOnly: true
+          # ports:
+          #   - name: http
+          #     containerPort: {{ .Values.service.port }}
+          #     protocol: TCP
+          # livenessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
+          # readinessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
```
I wouldn't like to deploy it to the cluster without this.

mvleandro replied:
Done!

mvleandro committed:
Adding health-check.

mvleandro committed:
Adding quotes on command-line parameters.

mvleandro committed:
Adding environment variables on deployment.

mvleandro committed:
Fixing environment variables names.

mvleandro committed:
Changing image tag.

mvleandro committed:
Updating keess image tag.

mvleandro committed:
Updating keess image tag.

mvleandro committed:
Stop creating namespace on chart.

Moumoutaru commented on charts/keess/templates/NOTES.txt:
```diff
@@ -0,0 +1 @@
+1. Keess deployed successfully.
```
What's the purpose of this file?

mvleandro replied:
This is the message displayed after the chart installation. It can be customized using variables, but It's enough for now.

Moumoutaru commented on charts/keess/templates/cluster-role-binding.yaml:
```diff
@@ -0,0 +1,16 @@
+- kind: ServiceAccount
+  name: {{ template "keess.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+{{- end }}
```
We should move Clusterroles/Rolebindings to PAC, no?

mvleandro replied:
We have the purpose to make keess public soon, so considering that the idea is to consolidate the deployment of all artifacts inside of this chart.

Moumoutaru replied:
Right that makes sense -- we may run into issues during deployment.

mvleandro replied:
I already tested the deployment on alpha and beta clusters. No issues until now.

Moumoutaru commented on charts/keess/templates/cluster-role.yaml:
```diff
@@ -0,0 +1,26 @@
+  resources:
+  - events
+  verbs: ["create"]
+{{- end }}
```
Same here. Move to PAC? @iblackman @bcdonadio ?

mvleandro replied:
Same before

iblackman replied:
The chart has the logic to not generated those if desired, which I consider correct. The proper way to deploy it in PAC can be addressed on the PR to deploy it in PAC.

iblackman replied:
Since he added the option (`rbac.create`) to not generate those it works for both scenarios. Now, how we will use this chart to deploy in pac is a discussion that we need to have in the PR in pac once we have it. For the chart it is good to have it as more generic and customizable as possible.

Moumoutaru commented on charts/keess/templates/secret.yaml:
```diff
@@ -0,0 +1,11 @@
+    {{- include "keess.labels" . | nindent 4 }}
+data:
+  config: {{ .Values.config.kubeconfigContent | trim | b64enc | quote }}
+{{- end }}
```
Newline to make git happy

mvleandro replied:
Done

mvleandro committed:
Adding new line at the end of files.

benlangfeld commented on charts/keess/values.yaml:
```diff
@@ -0,0 +1,104 @@
+# Declare variables to be passed into your templates.
+
+image:
+  repository: image-registry.powerapp.cloud/keess/keess
```
What publishes this image?

mvleandro replied:
@benlangfeld I just changed the chart to point to the image generated by Jenkins.

iblackman commented on charts/keess/templates/deployment.yaml:
```diff
@@ -0,0 +1,81 @@
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          args:
+          - run
```
Are you sure this works? Because the Dockerfile entrypoint already has `run` in it https://github.com/powerhome/keess/pull/2/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R15

mvleandro replied:
> Are you sure this works? Because the Dockerfile entrypoint already has `run` in it https://github.com/powerhome/keess/pull/2/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R15

You're correct. I forgot to change It here. I'm going to change this.

mvleandro replied:
Done!

iblackman commented on charts/keess/templates/deployment.yaml:
```diff
@@ -0,0 +1,81 @@
+          args:
+          - run
+          env:
+          - name: KEESS_SOURCE_CONTEXT
+            value: "{{ .Values.clusterName }}"
+          {{- if .Values.remoteClusters }}  
+          - name: KEESS_DESTINATION_CONTEXTS
+            value: "{{ .Values.remoteClusters }}"
+          {{- end }}
+          {{- if .Values.developmentMode }}
+          - name: KEESS_DEVELOPMENT_MODE
+            value: "{{ .Values.developmentMode }}"
+          {{- end }}
```
isn't easier to add those as `args`?
```suggestion
          args:
          - --sourceContext {{ .Values.clusterName }}
          {{- if .Values.remoteClusters }}  
          - --destinationContexts {{ .Values.remoteClusters }}
          {{- end }}
          {{- if .Values.developmentMode }}
          - --developmentMode {{ .Values.developmentMode }}
          {{- end }}
```

mvleandro replied:
Nope. I tried, and believe me: It's better to use envvars.

iblackman reviewed (commented):
If we decide to open source https://github.com/powerhome/keess and not [this repo](https://github.com/powerhome/helm-charts) we should probably consider to move this chart to the `keess` repo so people can have access to the chart.

benlangfeld commented:
> If we decide to open source https://github.com/powerhome/keess and not [this repo](https://github.com/powerhome/helm-charts) we should probably consider to move this chart to the `keess` repo so people can have access to the chart.

I think we should do this anyway.

mvleandro committed:
Updating application image and version.

mvleandro committed:
Removing unnecessary args.

mvleandro commented:
> I think we should do this anyway.

@benlangfeld  Could we do It in the near future project to open this project source?

benlangfeld commented:
> > I think we should do this anyway.
> 
> 
> 
> @benlangfeld  Could we do It in the near future project to open this project source?

Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove from this repo? Seems like a lot of work that is likely to not get done.

mvleandro commented:
> > > I think we should do this anyway.
> > 
> > 
> > @benlangfeld  Could we do It in the near future project to open this project source?
> 
> Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove from this repo? Seems like a lot of work that is likely to not get done.

The point is that the strategy we take is to divide it into parts. In this first moment we are treating the project as internal and later we will have a story just to deal with the issues necessary to open the project, so I think it is more cohesive to do this in this second moment.

mvleandro commented:
> > > > I think we should do this anyway.
> > > 
> > > 
> > > @benlangfeld  Could we do It in the near future project to open this project source?
> > 
> > 
> > Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove it from this repo? Seems like a lot of work that is likely to not get done.
> 
>

@benlangfeld  The point is that the strategy we took was to divide it into parts. In this first moment, we are treating the project as internal and later we will have a story just to deal with the issues necessary to open the project, so I think it is more cohesive to do this in this second moment.

benlangfeld commented:
> > > > I think we should do this anyway.
> > > 
> > > 
> > > @benlangfeld  Could we do It in the near future project to open this project source?
> > 
> > 
> > Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove from this repo? Seems like a lot of work that is likely to not get done.
> 
> The point is that the strategy we take is to divide it into parts. In this first moment we are treating the project as internal and later we will have a story just to deal with the issues necessary to open the project, so I think it is more cohesive to do this in this second moment.

I think we should prioritise reducing overall work and increasing the probability of overall success. What I'm suggesting costs us 15 minutes in the short term and saves hours in the medium term. I'm not asking for a large investment.

mvleandro commented:
> > > > > I think we should do this anyway.
> > > > 
> > > > 
> > > > @benlangfeld  Could we do It in the near future project to open this project source?
> > > 
> > > 
> > > Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove from this repo? Seems like a lot of work that is likely to not get done.
> > 
> > 
> > The point is that the strategy we take is to divide it into parts. In this first moment we are treating the project as internal and later we will have a story just to deal with the issues necessary to open the project, so I think it is more cohesive to do this in this second moment.
> 
> I think we should prioritise reducing overall work and increasing the probability of overall success. What I'm suggesting costs us 15 minutes in the short term and saves hours in the medium term. I'm not asking for a large investment.

@benlangfeld  I understand what you are asking, but the point is that your request is not consistent with the strategy we adopted in principle. Otherwise, we should be addressing all the points related to opening the project right now, which would slow down the delivery and increase the scope of the story.

benlangfeld commented:
> > > > > > I think we should do this anyway.
> > > > > 
> > > > > 
> > > > > @benlangfeld  Could we do It in the near future project to open this project source?
> > > > 
> > > > 
> > > > Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove from this repo? Seems like a lot of work that is likely to not get done.
> > > 
> > > 
> > > The point is that the strategy we take is to divide it into parts. In this first moment we are treating the project as internal and later we will have a story just to deal with the issues necessary to open the project, so I think it is more cohesive to do this in this second moment.
> > 
> > 
> > I think we should prioritise reducing overall work and increasing the probability of overall success. What I'm suggesting costs us 15 minutes in the short term and saves hours in the medium term. I'm not asking for a large investment.
> 
> @benlangfeld I understand what you are asking, but the point is that your request is not consistent with the strategy we adopted in principle. Otherwise, we should be addressing all the points related to opening the project right now, which would slow down the delivery and increase the scope of the story.

I am not asking you to do it for the purpose of open-sourcing the project. I am asking you to do it because it is cleaner. It just happens to *also* make the open-sourcing easier, but that is not at all the reason for my request. That is why I said "I think we should do this anyway", as in "independent of open-sourcing", rather than "I also think we should do this" or "I think we should do this now".

mvleandro commented:
> > > > > > > I think we should do this anyway.
> > > > > > 
> > > > > > 
> > > > > > @benlangfeld  Could we do It in the near future project to open this project source?
> > > > > 
> > > > > 
> > > > > Isn't it easier to do it now than to merge this, start using it, then have to come back later and migrate away from it and remove from this repo? Seems like a lot of work that is likely to not get done.
> > > > 
> > > > 
> > > > The point is that the strategy we take is to divide it into parts. In this first moment we are treating the project as internal and later we will have a story just to deal with the issues necessary to open the project, so I think it is more cohesive to do this in this second moment.
> > > 
> > > 
> > > I think we should prioritise reducing overall work and increasing the probability of overall success. What I'm suggesting costs us 15 minutes in the short term and saves hours in the medium term. I'm not asking for a large investment.
> > 
> > 
> > @benlangfeld I understand what you are asking, but the point is that your request is not consistent with the strategy we adopted in principle. Otherwise, we should be addressing all the points related to opening the project right now, which would slow down the delivery and increase the scope of the story.
> 
> I am not asking you to do it for the purpose of open-sourcing the project. I am asking you to do it because it is cleaner. It just happens to _also_ make the open-sourcing easier, but that is not at all the reason for my request. That is why I said "I think we should do this anyway", as in "independent of open-sourcing", rather than "I also think we should do this" or "I think we should do this now".

At no point am I judging whether it's more or less clean. Initially, even the chart was in the other repository, and I moved it here due to the strategy we adopted, in treating the project as closed now and then we open it. This conversation is unproductive and the problem keeps happening. And at the end of the day, the chart here or there will not change value delivery at all.

mvleandro commented:
@benlangfeld I created the PR moving the chart to the keess repository. Can you take a look there? [https://github.com/powerhome/keess/pull/5](https://github.com/powerhome/keess/pull/5)

mvleandro commented:
I'm archiving it because we are moving it to the keess repository.

mvleandro commented:
I'm reopening this PR because, after moving the chart to the KEESS repository, as it is still closed, the GIT utility asks for a login, which makes automation unfeasible at this point.

mvleandro commented:
I'm reopening this PR because, after moving the chart to the KEESS repository, as it is still closed, the GIT utility asks for a login, which makes automation unfeasible at this point.
